stubs/netaddr/netaddr/core.pyi

```diff
@@ -0,0 +1,39 @@
+from _typeshed import Incomplete
+from collections.abc import Iterator, Mapping
+from typing import TextIO
+from typing_extensions import Final
+
+BIG_ENDIAN_PLATFORM: bool
+P: Final = 1
+INET_PTON: Final = 1
+Z: Final = 2
+ZEROFILL: Final = 2
+N: Final = 4
+NOHOST: Final = 4
+
+class AddrFormatError(Exception): ...
+class AddrConversionError(Exception): ...
+class NotRegisteredError(Exception): ...
+
+def num_bits(int_val: int) -> int: ...
+
+class Subscriber:
+    def update(self, data: Incomplete) -> None: ...
+
+class PrettyPrinter(Subscriber):
+    fh: TextIO
+    write_eol: bool
+    def __init__(self, fh: TextIO = ..., write_eol: bool = ...) -> None: ...
```

Comment:
For IO types, we generally prefer to be as permissive as possible. In this case, it looks like `fh` doesn't _have_ to be a `TextIO` object; it can be anything with a `write()` method. We can use the `SupportsWrite` protocol from the `stdlib/_typeshed` directory to express this (you'll have to add `from _typeshed import SupportsWrite` at the top of the file):

```suggestion
    fh: SupportsWrite[str]
    write_eol: bool
    def __init__(self, fh: SupportsWrite[str] = ..., write_eol: bool = ...) -> None: ...
```